File: coding-test/programmers/src/kr/co/programmers/practice/CreatingJadenCaseStrings.java
package kr.co.programmers.practice;

public class CreatingJadenCaseStrings {

    public static void main(String[] args) {
        String s = "3people unFollowed me";
        System.out.println(new Solution().solution(s));
    }

    static class Solution {
        public String solution(String s) {
            String answer = "";

            s = s.toLowerCase();
            String[] wArr = s.split("");

            for(int i = 0; i < wArr.length; i++){

                if(i == 0){
                    answer += wArr[i].toUpperCase();
                    continue;
                }

                if(" ".equals(wArr[i-1])){
                    answer += wArr[i].toUpperCase();
                    continue;
                }

                answer += wArr[i];
            }

            return answer;
        }
    }
}

File: coding-test/nodejs/src/practice/creating-jadencase-strings.js
// @ts-check

/**
 * @param {string} s
 * @returns {string}
 */
function solution(s) {
  return s
    .split(' ')
    .map((x) => x.slice(0, 1).toUpperCase() + x.slice(1).toLowerCase())
    .join(' ')
}

const ss = ['3people unFollowed me', 'for the last week']
ss.forEach((s) => {
  console.log(solution(s))
})
